cmd/fund/fund.go

```diff
@@ -0,0 +1,132 @@
+package fund
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"log"
+	"math/big"
+
+	_ "embed"
+
+	"github.com/chenzhijie/go-web3"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/spf13/cobra"
+)
+
+var (
+	//go:embed usage.md
+	usage                string
+	walletCount          int
+	fundingWalletAddress string
+	fundingWalletPK      string
+	chainID              int
+	chainRPC             string
+	walletFundingAmt     float64
+	walletFundingGas     uint64
+)
+
+func generateWalletAddresses(numWallets int) ([]*common.Address, error) {
+	var addresses []*common.Address
+
+	for i := 0; i < numWallets; i++ {
+		account, err := crypto.GenerateKey()
+		if err != nil {
+			log.Fatal(err)
+		}
+		addr := crypto.PubkeyToAddress(account.PublicKey)
+		addresses = append(addresses, &addr)
+	}
+	return addresses, nil
+}
+
+func fundWallets(web3Client *web3.Web3, wallets []*common.Address, senderAddress common.Address, senderPrivateKey *ecdsa.PrivateKey, amountWei *big.Int, walletFundingGas uint64) error {
+	nonce, err := web3Client.Eth.GetNonce(senderAddress, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for _, wallet := range wallets {
+		_, err := web3Client.Eth.SyncSendRawTransaction(
```

Comment:
I'm not familiar with `web3Client` library but I think we could improve this fn by sending batch calls. I know this is something you can do with `github.com/ethereum/go-ethereum/rpc`.

https://pkg.go.dev/github.com/ethereum/go-ethereum@v1.13.2/rpc#Client.BatchCallContext